server/reload.go

```diff
@@ -250,9 +250,25 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
 			diffOpts = append(diffOpts, &authTimeoutOption{newValue: newValue.(float64)})
 		case "users":
 			diffOpts = append(diffOpts, &usersOption{newValue: newValue.([]*User)})
+		case "port":
+			// check to see if newValue == 0 and continue if so.
+			if newValue == 0 {
+				// ignore RANDOM_PORT
+				continue
+			} else {
+				return nil, fmt.Errorf("Config reload not supported for %s: old=%v, new=%v",
+					field.Name, oldValue, newValue)
+			}
+		case "http_port", "https_port":
+			// check to see if newValue == -1 (RANDOM_PORT for http/https monitoring port)
+			if newValue == -1 {
```

Comment:
I don't think this is doing what you think it's doing. Those strings are the field names as they appear in the `Options` struct lowercased, so those would actually be `httpport` and `httpsport`, which means that case isn't doing anything AFAICT. It looks like this is only needed for `port`.